### DISCUSSION TITLE
atomWithDefault should execute default read function on reset even if it haven't changed before

### DISCUSSION THREAD

**Discussion by Nurai1** [pmndrs/jotai#1747] [Q&A] [answered]
What happens now?

When you reset atomWithDefault but u never change it at runtime atom just won't execute default read function.

What I want it to be? 

atomWithDefault should execute default read function on reset always

Why?

I have async cached atomWithDefault and I need to have a default request each time some data relates to it were updated. And I just use data from cache and when I need a new request, I only need to reset it and that's all. But now I set a value before reset like this.

`			await set(companiesFamilyAsyncAtom(currentPageParams), {})
			await set(companiesFamilyAsyncAtom(currentPageParams), RESET)
`

Family because I use it for pagination. The main point that here is one unnecessary line of code. I hope somebody answer on this, thank you foir reading.

**Comment by dai-shi** [comment, marked as answer]
It's not reset, in our api. It's refresh. Check out atomWithRefresh.

**Reply by Nurai1** [reply]
@dai-shi It fits better, thank you! BTW I need to add _atomWithRefresh_ manually, right?

**Reply by dai-shi** [reply]
Yeah, it's a pattern, not a provided util.
